### PR TITLE
Fix rh side ad slot for immersive content

### PR DESCRIPTION
**Pull request by adamnfish**
## What does this change?

Adds logic to the Right Hand Side adslot code to handle immersive content. This came up from photo essays, but actually applies more broadly than that (immersive content).

If the content is "immersive" we measure the distance between the top of the article body and the first immersive element. We then change the ad sizes that are eligible for the RH-Side ad slot to ensure that there will be enough space for the ad to display.
Pixels always fit so we allow 1x1 and 2x2 ads in the slot in all cases.

We also remove `sticky` from the ad slot in immersive cases because a sticky ad might need more space. This means we don't need to do more complex calculations and adjust the sticky logic to take immersive element position into account.

Commercial ticket:
https://trello.com/c/1GKP3WUy/53-ads-overlapping-images-on-photo-essays

## Screenshots

Using the new `forceads` flag, we can see what this looks like for an example piece of immersive content in production.

immersive-rh-ad-bug

This is what it looks like in dev with this change applied.

immersive-rh-ad-fix

## What is the value of this and can you measure success?

Up to now the only workaround has been to manually disable ads on any piece of content that is affected. This is laborious and easy to forget. More importantly, it disables ads for the entire piece of content, not just for this slot. There may well be space in the article for ads and certainly, the comments will be unaffected.

Success will be the ad revenue generated by existing pieces of immersive content once ads can be re-enabled, as well as for all future immersives. It will also be the time and effort saved for Commercial and CP dealing with ads overlapping immersive content.

## Checklist

Should probs run this by commercial people and give CP a heads-up. There may be bugs when we test it on more diverse content though.

### Does this affect other platforms?

Nope, this is a website platform change.

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js [hunk 1]
```diff
@@ -3,14 +3,20 @@ import $ from 'lib/$';
...
-    config.page.isImmersive ? minImmersiveArticleHeight : minArticleHeight;
+const calculateAllowedAdSlots = (availableSpace: number): string => {
+    if (availableSpace > 600) {
+        return '1,1|2,2|300,250|300,274|300,600|fluid';
+    } else if (availableSpace > 274) {
+        return '1,1|2,2|300,250|300,274';
+    } else if (availableSpace > 250) {
+        return '1,1|2,2|300,250';
+    }
+    return '1,1|2,2';
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: What is a 2x2?

**Review by kelvin-chappell — APPROVED** [review]
Good solution! 👍

**Comment by adamnfish** [comment]
Update:

@paperboyo also pointed out that some articles use a "no-main-media-first-pic-immersed" layout. This PR also fixes the ads in these cases.
Here's an example.
https://www.theguardian.com/artanddesign/2019/jan/12/buy-guardian-classic-photograph-early-morning-frost?forceads

In this case there can be an ad in the article, but the RH-Side slot appears over the picture so ads have been disabled for this content.

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.js (hunk 1 above)
Comment (adamnfish) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.js]: Good question! It was there before so I've left it in, but I have no idea whether that would ever appear in practice...

**Pull request merged**